jstreegrid.js

```diff
@@ -857,16 +934,30 @@
 			}
 			return result;
 		};
-		this._hide_grid = function (node) {
-			var children = node && node.children_d ? node.children_d : [], i;
-			// go through each column, remove all children with the correct ID name
-			findDataCell(this.uniq,children,this._gridSettings.gridcols).remove();
+    this._hide_grid = function (node) {
+      if (!node) { return true; }
+      this._detachColumns(node.id);
+      var children = node.children ? node.children : [], cols = this._gridSettings.columns || [], treecol = this._gridSettings.treecol;
+      // try to remove all children
+      for (var i = 0, len = children.length; i < len; i++) {
```

Comment:
Is this deep nested loop necessary? How does this new `_hide_grid()` differ from the original, which just did `findDataCell(this.uniq,children,this._gridSettings.gridcols).remove();`?